{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96371f5a-5455-4f09-a3fa-32e9206c5f01",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import yagmail\n",
    "import time\n",
    "import schedule\n",
    "from sqlalchemy import create_engine\n",
    "\n",
    "engine=create_engine('postgresql://postgres:password@localhost/dvdrental')\n",
    "db=engine.connect()\n",
    "user=yagmail.SMTP(user='1030297@cuea.edu',password='cuea2020')\n",
    "df=pd.read_sql('select * from payment',db)\n",
    "(df).to_excel(r'C:\\Output\\tester.xlsx')\n",
    "user.send(to='vicmose.vm@gmail.com',subject='test',contents='test',attachments=r'C:\\Output\\tester.xlsx')\n",
    "db.close()\n",
    "\n",
    "quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7ed96945-9cee-4164-984b-5893f9e28477",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# def mailer():\n",
    "#     df=pd.read_sql('select * from payment',db)\n",
    "#     (df).to_excel(r'C:\\Output\\tester.xlsx')\n",
    "#     user.send(to='vicmose.vm@gmail.com',subject='test',contents='test',attachments=r'C:\\Output\\tester.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b01dd3d-bc57-4043-a445-0d8fa75ff40c",
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mCanceled future for execute_request message before replies were done"
     ]
    },
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. View Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "# schedule.every(50).seconds.do(mailer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "036879aa-0d7e-4af3-afd4-0057a7651921",
   "metadata": {},
   "outputs": [],
   "source": [
    "# while True:\n",
    "#     schedule.run_pending()\n",
    "#     time.sleep(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7dc8469b-ac31-42d0-9ec3-adac6952138d",
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. View Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3d9d026",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
